src/Clapi/RelayApi.hs

```diff
@@ -118,6 +118,21 @@ relayApiProto ownerMv selfAddr =
         newOwnerMap <- lift (readMVar ownerMv)
         pubOwnerMap oldOwnerMap newOwnerMap
         return newOwnerMap
+    clientPov ci i (FRBClient (UpdateBundle errs oums)) = let
+        theirTime = Map.lookup (pathSegmentFor i) ci
+        smt (Just a) (Just b) = a - b
```

Comment:
```haskell
import Control.Applicative (liftA2)
smt = liftA2 (-)
```
is more canonical for lifting stuff. I keep wanting to do stuff more like this, rather than leaving incomplete pattern matches as kinda time bombs everywhere (although I understand your motivation for making it fail hard on the line where you got the weird thing, rather than keeping limping on with weird data):
```haskell
toMonoid :: (Applicative f, Monoid (f a)) => Maybe a -> f a
toMonoid Nothing = mempty
toMonoid (Just a) = pure a

relClientINfo [td] = fmap toClapiValue . toMonoid $ smt (fromClapiValue td) theirTime
```

Anyway, probably a discussion for another day.

Reply:
That type checks, and will serialise but with something that doesn't match the type info, so I don't know if that's better or worse.

Reply:
Probably not worth it, was just exploring the possibility. Also, was nice to write my cleanest definition of toMonoid to date :)